src/GitVersionCore/Model/BranchCommit.cs

```diff
@@ -7,14 +7,14 @@ namespace GitVersion
     {
         public static readonly BranchCommit Empty = new BranchCommit();
 
-        public BranchCommit(Commit commit, Branch branch) : this()
+        public BranchCommit(ICommit commit, IBranch branch) : this()
         {
             Branch = branch;
             Commit = commit;
```

Comment:
Is `null` a desirable value for `commit`  and `branch` or can we add `null` guards here?

Reply:
I will review this on next PR